### PR TITLE
Phase 3

### PR DESCRIPTION
**Pull request by ashwin8448**
# hrmApp

**New features:**
* Pagination addded to web application.

**Storage:**
* The application uses localStorage.

**Upcoming features:**
* Search skills in the filter box.
* Skills field in new employee form will be searchable.

### CODE REVIEW

**Review by vipinmsqburst — CHANGES_REQUESTED** [review]

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  pagination,
+  changePageNumber,
+} from "./state.js";
+import { submitValidation, validationReset } from "./formValidation.js";
+import { toastHandler } from "./toast.js";
+import { loadEmployeeData } from "./edit.js";
+import { employees, setEmployees } from "./firebase.js";
+import { filteredEmployees } from "./filter.js";
+
+let employeesDb = [
```
[the remark this answers is not in the document]
Comment (ashwin8448) [reply js/main.js:40]: Now I am using localStorage for handling data. So I hard coded few employee details to check whether my functionalities are working correctly.

[the remark this answers is not in the document]
Comment (vipinmsqburst) [reply js/main.js:40]: Ok

File: js/handlers.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  else {
+    if (target.classList.contains("filter-checkbox")) {
+      if (!state.filterBy.skills.includes(target.value)) {
+        selectedSkills.innerHTML += `<div class="selected-skill-button ${target.value} flex">
+            <span>${target.value}</span>
+            <button type="button" class="skill-close"><img data-skill="${target.value}" src="./assets/images/close_button_icon.svg" alt="close icon"></button>
+            </div>`;
+        state.filterBy.skills.push(target.value);
+      } else {
+        console.log(target.value)
```
Comment (dineshkumarrajaguru) [inline js/handlers.js]: Please remove console.log

File: js/handlers.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  tempEmployee.department = newEmployeeForm["department"].value;
+  tempEmployee.role = newEmployeeForm["role"].value;
+  for(let checkbox of document.querySelectorAll(".form-checkbox"))
+  {
+    if(checkbox.checked==true){
+      skills.push(checkbox.value)
+    }
+  }
+  tempEmployee.skills = skills
+  console.log(tempEmployee.skills)
```
Comment (dineshkumarrajaguru) [inline js/handlers.js]: Remove console.

**Review by dineshkumarrajaguru — CHANGES_REQUESTED** [review]

File: js/displayTable.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  overlay,
+  pageCustomInput,
+  totalPages,
+} from "./elements.js";
+
+export const displayTable = () => {
+  loaderContainer.classList.remove("open");
+  overlay.classList.remove("open");
+  filterEmployees();
+  setTotalPages(filteredEmployees.length);
```
[the remark this answers is not in the document]
Comment (ashwin8448) [reply js/displayTable.js:16]: filteredEmployees.length is used to calculate the total number of pages.

File: js/edit.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { newEmployeeForm, viewEmployee } from "./elements.js";
+import { state } from "./state.js";
+import { filterEmployees, filteredEmployees } from "./filter.js";
+
+export const loadEmployeeData = (id, mode) => {
+  state.filterBy.id.push(id);
+  filterEmployees();
```
[the remark this answers is not in the document]
Comment (ashwin8448) [reply js/edit.js]: Modified filterEmployees to filter a given array and return filtered array.

File: js/edit.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    newEmployeeForm["email"].value = employee.email;
+    newEmployeeForm["doj"].value = employee.doj;
+    newEmployeeForm["department"].value = employee.department;
+    newEmployeeForm["role"].value = employee.role;
+    for(let skill of employee.skills){
+      formOptionsContainer.querySelector(`[data-form-skill="${skill}"]`).checked=true;
+    }
+  } else {
+    viewEmployee.children[2].firstElementChild.innerHTML =
+      employee.fname + " " + employee.lname;
```
Comment (dineshkumarrajaguru) [inline js/edit.js:24]: In future if someone changes the order of HTML element then we will end up with wrong data mapping.

**Pull request merged**